docs/reference/specifications/providers.md

```diff
@@ -0,0 +1,251 @@
+| tls                | FLAGD_TLS                  | connection encryption                                                           | boolean                  | false                         | rpc & in-process    |
+| socketPath         | FLAGD_SOCKET_PATH          | alternative to host port, unix socket                                           | String                   | null                          | rpc & in-process    |
+| certPath           | FLAGD_SERVER_CERT_PATH     | tls cert path                                                                   | String                   | null                          | rpc & in-process    |
+| deadline           | FLAGD_DEADLINE_MS          | deadline for unary calls, and timeout for initialization                        | int                      | 500                           | rpc & in-process    |
+| streamDeadlineMs   | FLAGD_STREAM_DEADLINE_MS   | deadline for streaming calls, useful as an application-layer keepalive          | int                      | 600000                        | rpc & in-process    |
+| retryBackoffMs     | FLAGD_RETRY_BACKOFF_MS     | initial backoff for stream retry                                                | int                      | 1000                          | rpc & in-process    |
+| retryBackoffMaxMs  | FLAGD_RETRY_BACKOFF_MAX_MS | maximum backoff for stream retry                                                | int                      | 120000                        | rpc & in-process    |
+| retryGraceAttempts | FLAGD_RETRY_GRACE_ATTEMPTS | amount of stream retry attempts before provider moves from STALE to ERROR state | int                      | 5                             | rpc & in-process    |
+| keepAliveTime      | FLAGD_KEEP_ALIVE_TIME_MS   | http 2 keepalive                                                                | long                     | 0                             | rpc & in-process    |
+| cache              | FLAGD_CACHE                | enable cache of static flags                                                    | String - lru, disabled   | lru                           | rpc                 |
+| maxCacheSize       | FLAGD_MAX_CACHE_SIZE       | max size of static flag cache                                                   | int                      | 1000                          | rpc                 |
+| selector           | FLAGD_SOURCE_SELECTOR      | selects a single sync source to retrieve flags from only that source            | string                   | null                          | in-process          |
+| contextEnricher    | -                          | sync-metadata to evaluation context mapping function                            | string                   | identity function             | in-process          |
```

Comment:
This is a new param. Already existing in the Java provider.

Reply:
I had to look this one up in the Java impl to understand it. 
Is it really a `string` type? I'd classify this as `in-code configuration` or `function` or `callback`.

Reply:
ya it should definitely be function.

Reply:
fixed.